### ISSUE TITLE
[bug] Using cmake_layout breaks cmake_paths generator

### ISSUE THREAD

**Issue by Richard-W** [conan-io/conan-center-index#12149]
### Environment Details
  * os: Ubuntu 22.04
  * gcc version: 11.2
  * conan version: 1.51.1
  * python version: 3.10.4

### Steps to reproduce

In an empty directory create a `conanfile.txt` with the following content:
```
[requires]
fmt/8.1.1

[generators]
cmake_paths
```
Issue `conan install .` and inspect `conan_paths.cmake`:
```
set(CONAN_FMT_ROOT "<path_to_conan_cache>/fmt/8.1.1/_/_/package/3bb43d390932310d9ef00e9986e41f02509d79e8")
set(CMAKE_MODULE_PATH  ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_LIST_DIR})
set(CMAKE_PREFIX_PATH  ${CMAKE_PREFIX_PATH} ${CMAKE_CURRENT_LIST_DIR})
```

**Expectation**: `CMAKE_MODULE_PATH` and `CMAKE_PREFIX_PATH` contain the value of `CONAN_FMT_ROOT`
**Actual**: Neither variable contains the value of `CONAN_FMT_ROOT`.

### Possible cause

The `cmake_paths` generator generates the `CMAKE_*` variables by concatenating the `build_paths` arrays of the dependencies. The default value for this path (which would be the correct value here) is overriden in `_call_package_info` in `conans/client/installer.py` if `hasattr(conanfile, "layout")` is true.

The correct value is set in `conan/client/installer.py:659`:
```
conanfile.cpp_info = CppInfo(conanfile.name, package_folder)
```

But it is later overridden in `conan/client/installer.py:679`:
```
if hasattr(conanfile, "layout"):
    # Old cpp info without defaults (the defaults are in the new one)
    conanfile.cpp_info = CppInfo(conanfile.name, package_folder,
                                 default_values=CppInfoDefaultValues())
```

**Comment by SpaceIm** [comment]
I don't understand, fmt doesn't contain build scripts. It's a conan bug of cmake_paths, not fmt recipe bug.

**Comment by lasote** [comment]
@SpaceIm, not at all, maybe [Richard-W](https://github.com/Richard-W) has more information about his usage of fmt and why he needs the `CMAKE_MODULE_PATH and CMAKE_PREFIX_PATH containing the CONAN_FMT_ROOT`

**Comment by SpaceIm** [comment]
According to https://docs.conan.io/en/latest/reference/generators/cmake_paths.html, it looks like a bug. In the file generated by `cmake_paths`, `CMAKE_MODULE_PATH` & `CMAKE_PREFIX_PATH` should contain root package folders of dependencies.

**Comment by Richard-W** [comment]
@lasote Context is a project that relies on `CMAKE_MODULE_PATH` to find its dependencies. This cannot be easily changed since the project also needs to build in somewhat exotic environments where conan is unavailable. Using `cmake_paths` is a good way to create conan packages of that project without having to touch its build system.

**Comment by SpaceIm** [comment]
> @lasote Context is a project that relies on `CMAKE_MODULE_PATH` to find its dependencies.

`CMAKE_PREFIX_PATH` I guess. root package folder of fmt in `CMAKE_MODULE_PATH` is useless since there is no cmake module file. `CMAKE_MODULE_PATH` is not used at all in commands like find_file, find_program, find_library etc.

**Comment by lasote** [comment]
Transfering back to conan.
I have to discuss what to do with this issue. The documentation at [here](https://docs.conan.io/en/latest/reference/generators/cmake_paths.html) was wrong, `cmake_paths` never used the root of the packages but the `builddirs`, and it was the root folder the default for `builddirs` until, when layout was declared, we moved in [this direction](https://github.com/conan-io/conan/pull/10186).

So now I have to discuss if we should "fix" the `cmake_paths` generator (not maintained anymore) by adding the root folder always. Tagging it as `look_into`.
